conjur/user/user_logic.py

```diff
@@ -0,0 +1,100 @@
+# -*- coding: utf-8 -*-
+
+"""
+UserLogic module
+
+This module is the business logic for handling all user-related activity
+"""
+
+# Builtins
+import logging
+
+# Internals
+from conjur.credentials_data import CredentialsData
+
+class UserLogic:
+    """
+    UserLogic
+
+    This class holds the business logic for handling
+    user activity
+    """
+    def __init__(self, conjurrc_data, credentials_from_store, client, resource):
+        self.conjurrc_data=conjurrc_data
+        self.credentials_from_store=credentials_from_store
+        self.client=client
+        self.resource=resource
+
+    # pylint: disable=logging-fstring-interpolation
+    def rotate_api_key(self, resource_to_update):
+        """
+        Method to trigger two types of API key actions.
+        1. The user can either rotate their own API key or
+        2. Rotate another users API key
+        """
+        loaded_credentials = self.extract_credentials_from_credential_store()
+        user_from_credential_store = loaded_credentials['login_id']
+        if resource_to_update == user_from_credential_store:
+            raise Exception("Error: To rotate the API key of the currently logged-in user "\
+                            " use this command without any flags or options")
```

Comment:
I think that user instructions should be created in the controller and the logic should throw logic exceptions. also it is recommended to be as much as specific as you can when throwing an error so InvalidOperationException is better. if you can't find the type of exception you want to throw you can allways create one. it will be good if we have exceptions module for the CLI

Reply:
Agreed! but this logic is appropriate only here because we have extracted who the logged in user is. How would you get around this?

I agree I should make this a CLI-specific error. Will do

Reply:
how would you call this custom error?
I was thinking `RotateApiKeyUserLoggedInUserSameAsFlag` but that is very very long

Reply:
for the first comment - throw here and catch in the controller. there you can set the user message

for the second - i think this is too specific I don't think you will use `RotateApiKeyProvidedUserIsSameAsLoggedIn ` 
 somewhere else (if you think I'm wrong than it is indeed good exception).
 this looks like invalidOperation so what about `invalidOperationException`? this shows what is the reason for the exception + you can specify the exact reason in the string inside.

now the controller can have a 

`try: .. 
except invalidOperationException: handle specific error here and show the user the desired message 
except Exception: handle general error that you didn't see coming`

Reply:
> for the first comment - throw here and catch in the controller. there you can set the user message

if I am understanding, you are suggesting I throw the InvalidOperation from the logic and intercept and rethrow from the controller? So it looks like:
```
# logic.py

raise InvalidOperation

# controller.py
try:
..
except InvalidOperation
...
except Exception
...

Reply:
![exactly](https://user-images.githubusercontent.com/76062747/105689600-a2743f00-5f03-11eb-9f29-6f58027a0471.gif)